occlusion.html

```diff
@@ -0,0 +1,708 @@
+        gl.enableVertexAttribArray(1);
+
+        normalBuffer = gl.createBuffer();
+        gl.bindBuffer(gl.ARRAY_BUFFER, normalBuffer);
+        gl.bufferData(gl.ARRAY_BUFFER, sphere.normals, gl.STATIC_DRAW);
+        gl.vertexAttribPointer(2, 3, gl.FLOAT, false, 0, 0);
+        gl.enableVertexAttribArray(2);
+
+        indices = gl.createBuffer();
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, indices);
+        gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, sphere.indices, gl.STATIC_DRAW);
+
+        gl.bindVertexArray(null);
+
+        sphere.bbox = utils.computeBoundingBox(sphere.positions, {geo: true});
+
+        var sphereBBoxArray = gl.createVertexArray();
+        gl.bindVertexArray(sphereBBoxArray);
+
+        positionBuffer = gl.createBuffer();
+        gl.bindBuffer(gl.ARRAY_BUFFER, positionBuffer);
+        gl.bufferData(gl.ARRAY_BUFFER, sphere.bbox.bbox.positions, gl.STATIC_DRAW);
+        gl.vertexAttribPointer(0, 3, gl.FLOAT, false, 0, 0);
+        gl.enableVertexAttribArray(0);
+
+        gl.bindVertexArray(null);
+
+        var NUM_NODES = 512;
+        var NUM_PER_ROW = 8;
+        var RADIUS = 0.6;
+        var nodes = new Array(NUM_NODES);
+
+        for (var i = 0; i < NUM_NODES; ++i) {
+            var angle = 2 * Math.PI * (i % NUM_PER_ROW) / NUM_PER_ROW;
+            var x = Math.sin(angle) * RADIUS + 2 * Math.floor(i / 128) - 0.5;
+            var y = (Math.floor(i / NUM_PER_ROW) % 4) / (NUM_PER_ROW / 4) - 0.75;
+            var z = Math.cos(angle) * RADIUS - 2 * Math.floor(i / 32 % 4);
+
+            if (i % 2 == 0) {
+                // sphere low
+                nodes[i] = {
+                    scale: [0.8, 0.8, 0.8],
+                    rotate: [0, 0, 0], // Will be used for global rotation
+                    translate: [x, y, z],
+                    modelMatrix: mat4.create(),
+
+                    vao: sphereArray,
+                    indicesLength: sphere.indices.length,
+
+                    bbox: sphere.bbox,
+                    vaobbox: sphereBBoxArray,
+                    bboxNumVertices: sphere.bbox.bbox.positions.length / 3,
+
+                    query: gl.createQuery()
+                };
+            } else {
+                // sphere high
+                nodes[i] = {
+                    scale: [0.8, 0.8, 0.8],
+                    rotate: [0, 0, 0], // Will be used for global rotation
+                    translate: [x, y, z],
+                    modelMatrix: mat4.create(),
+
+                    vao: sphereHighArray,
+                    indicesLength: sphereHigh.indices.length,
+
+                    bbox: sphereHigh.bbox,
+                    vaobbox: sphereHighBBoxArray,
+                    bboxNumVertices: sphereHigh.bbox.bbox.positions.length / 3,
+
+                    query: gl.createQuery()
+                };
+            }
+            
+        }
+
+        nodes.sort(function(a, b){
+            var z = a.translate.z - b.translate.z;
```

Comment:
I'm thinking just transform into to view space, to make it obvious this is a depth sort.